tests/components/madvr/conftest.py

```diff
@@ -0,0 +1,67 @@
+"""MadVR conftest for shared testing setup."""
+
+from collections.abc import Generator
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import pytest
+
+from homeassistant.const import CONF_HOST, CONF_PORT
+
+from .const import CONFIG_ENTRY, MOCK_CONFIG, MOCK_MAC
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+def mock_setup_entry() -> Generator[AsyncMock, None, None]:
+    """Override async_setup_entry."""
+    with patch(
+        "homeassistant.components.madvr.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        yield mock_setup_entry
+
+
+@pytest.fixture
+def mock_madvr_client() -> Generator[AsyncMock, None, None]:
+    """Mock a MadVR client."""
+    with patch(
+        "homeassistant.components.madvr.config_flow.Madvr", autospec=True
+    ) as mock_client:
```

Comment:
Let's also patch the Madvr in the normal code, check `airgradietn` on how we patch it there